src/client/interpreter/configuration/interpreterSelector.ts

```diff
@@ -51,6 +56,24 @@ export class InterpreterSelector implements IInterpreterSelector {
         interpreters.sort(this.interpreterComparer.compare.bind(this.interpreterComparer));
         return Promise.all(interpreters.map(item => this.suggestionToQuickPickItem(item, resource)));
     }
+
+    protected async resetInterpreter() {
+        const setInterpreterGlobally =
+            !Array.isArray(this.workspaceService.workspaceFolders) ||
+            this.workspaceService.workspaceFolders.length === 0;
+        let configTarget = ConfigurationTarget.Global;
+        let wkspace: Uri | undefined;
+        if (!setInterpreterGlobally) {
+            const targetConfig = await this.getWorkspaceToSetPythonPath();
+            if (!targetConfig) {
+                return;
+            }
+            configTarget = targetConfig.configTarget;
+            wkspace = targetConfig.folderUri;
+        }
+
+        await this.pythonPathUpdaterService.updatePythonPath(undefined, configTarget, 'ui', wkspace);
+    }
```

Comment:
This would be simpler if `this.getWorkspaceToSetPythonPath()` (AKA `getConfigTarget()`) returned a proper value for global:

```suggestion
    protected async resetInterpreter() {
        const targetConfig = await this.getConfigTarget();
        if (!targetConfig) {
            return;
        }
        await this.pythonPathUpdaterService.updatePythonPath(undefined, targetConfig.configTarget, 'ui', targetConfig.folderUri);
    }
```

Along those lines, I recommend replacing line 169 (`return undefined` in `getWorkspaceToSetPythonPath()`) with the following:

```typescript
            return {
                folderUri: undefined,
                configTarget: ConfigurationTarget.Global
            }
```

That is also better because it allows us to distinguish from the canceled-quickpick case.

Reply:
Agreed!